src/cordova/app.cordova.ts

```diff
@@ -31,6 +35,58 @@ function onDeviceReady() {
   setupLinkListener()
 
   document.addEventListener("backbutton", event => contentWindow.postMessage({ id: "backbutton" }, "*"), false)
+
+  isAuthenticationAvailable()
+    .then(startAuthentication)
+    .catch(hideSplashScreen)
+}
+
+function getClientSecret(contentWindow: Window) {
+  return new Promise<string>((resolve, reject) => {
+    initializeStorage(contentWindow).then(secureStorage => secureStorage.get(resolve, reject, storeKeys.clientSecret))
+  })
+}
+
+function startAuthentication() {
+  lastAuthenticationTimestamp = Date.now()
+
+  const contentWindow = iframe.contentWindow
+  if (contentWindow) {
+    getClientSecret(contentWindow).then(clientSecret => {
+      showDialogue(clientSecret)
+    })
+  }
+}
+
+function showDialogue(clientSecret: string) {
+  showAuthenticationDialogue(clientSecret)
+    .then(hideSplashScreen)
+    .catch(exitApp)
+}
+
+function exitApp() {
+  window.navigator.app.exitApp()
```

Comment:
The `app` type is not available on `window.navigator` so we have to define it somehow somewhere.